source/mir/random/flex/internal/calc.d

```diff
@@ -0,0 +1,149 @@
+module mir.random.flex.internal.calc;
+
+import mir.random.flex.internal.types : Interval;
+
+/**
+Calculate the mean between two points using the arcmean:
+
+    tan(0.5 * (atan(l) + atan(r))
+
+In contrast to the normal mean (`0.5 * (l + r)`) being a geometric plane,
+the arcmean favors the mean region more.
+
+Params:
+    l = Left point
+    r = Right point
+
+Returns:
+    Splitting point within the interval
+
+See_Also:
+    $(LINK2 http://www.wolframalpha.com/input/?i=tan(0.5+*+(ArcTan%5Bx%5D+%2B+ArcTan%5By%5D)),
+    WolframAlpha visualization of the arc-mean)
+
+References:
+    Hormann, W., J. Leydold, and G. Derflinger.
+    "Automatic Nonuniform Random Number Generation." (2004): Formula 4.23
+*/
+auto arcmean(S, bool sorted = false)(const S l, const S r)
+{
+    import std.math: atan, tan;
+
+    S x = l;
+    S y = r;
+    static if (!sorted)
+    {
+        if (y < x)
+        {
+            S t = x;
+            x = y;
+            y = t;
+        }
+    }
```

Comment:
this doesn't make sense. An `Interval` guarantees `lx < rx`.
@9il can I change the api to a `Interval` or move it somewhere else if it's thought to be more generic?